### PR TITLE
Node 18 Update

### PR DESCRIPTION
**Pull request by fajar-apri-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #38, #39

## Summary:
#39 fixes is included in #38 

## Type of change:

Please delete options that are not relevant.

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -17,68 +17,62 @@
...
-    "@alaskaairux/icons": "^4.7.0",
-    "@webcomponents/webcomponentsjs": "^2.5.0",
-    "focus-visible": "^5.2.0"
+    "@alaskaairux/icons": "^4.7.0"
   },
   "devDependencies": {
-    "@alaskaairux/auro-icon": "^3.4.0",
-    "@alaskaairux/design-tokens": "^3.12.2",
-    "@alaskaairux/icons": "^4.21.2",
+    "@alaskaairux/auro-icon": "^3.4.4",
```
Comment (blackfalcon) [inline package.json:28]: Deprecated. Update to new version. 

https://www.npmjs.com/package/@alaskaairux/auro-icon

File: package.json [hunk 2]
```diff
@@ -133,7 +127,7 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render dist:js build:api bundler postinstall test",
+    "build": "npm-run-all build:sass sass:render dist:js build:api bundler postinstall test types",
     "build:ci": "npm-run-all sweep build",
     "build:api": "wca analyze 'src/auro-avatar.js' --outFiles docs/api.md",
     "build:demo": "npm-run-all build demo:rm:build demo:new:build demo:copy:index demo:copy:demo demo:update:index",
```
Comment (blackfalcon) [inline package.json:133]: Delete this script

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (fajar-apri-alaska) [reply package.json:28]: on it

File: package.json (hunk 2 above)
Comment (fajar-apri-alaska) [reply package.json:133]: build:demo one? sure.

File: package.json [hunk 3]
```diff
@@ -133,34 +127,29 @@
...
     "bundler": "rollup -c",
     "bundler:test": "rollup -c -w",
     "scssLint": "stylelint \"./src/**/*.scss\"",
-    "demo:copy:demo": "copyfiles -u 1 -V './demo/demo.md' ./build/demo",
-    "demo:copy:index": "copyfiles -u 1 -V './demo/index.html' ./build",
-    "demo:new:build": "mkdir ./build && mkdir ./build/css && mkdir ./build/demo",
-    "demo:rm:build": "rm -rf ./build",
-    "demo:update:index": "node ./scripts/prepForBuild",
-    "dev": "npm run build:watch",
+    "dev": "concurrently --kill-others 'npm run build:watch' 'npm run serve'",
```
Comment (fajar-apri-alaska) [inline package.json:140]: Last changes I forgot to make the run dev in a single terminal only, so I have added it now.

File: package.json [hunk 4]
```diff
@@ -17,68 +17,62 @@
   "main": "index.js",
   "license": "Apache-2.0",
   "dependencies": {
-    "chalk": "^4.1.2",
-    "lit-element": "^2.5.1"
+    "@aurodesignsystem/auro-icon": "^4.0.2",
```
Comment (fajar-apri-alaska) [inline package.json:20]: Also move the auro-icon to dependencies since it is in the src file.
Let me know if this is purposely not needed, @blackfalcon.

Comment (blackfalcon) [reply package.json:20]: Please move that to `peerDependencies`

Comment (fajar-apri-alaska) [reply package.json]: done~

File: .husky/pre-commit [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
 
-./node_modules/.bin/npm-run-all preCommit test linters postinstall
+./node_modules/.bin/npm-run-all linters test
```
Comment (blackfalcon) [inline .husky/pre-commit:4]:
```suggestion
./node_modules/.bin/npm-run-all test linters postinstall
```

File: .stylelintrc [hunk 6]
```diff
@@ -1,119 +1,10 @@
 {
-  "extends": ["stylelint-config-recommended", "stylelint-config-idiomatic-order"],
-  "plugins": [
-    "stylelint-scss"
-  ],
+  "extends": "stylelint-config-recommended",
```
Comment (blackfalcon) [inline .stylelintrc:2]: This is wrong. See https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.stylelintrc

File: README.md [hunk 7]
```diff
@@ -42,24 +42,28 @@ import "@aurodesignsystem/auro-avatar";
...
 
-### Bundle example code
-
-**NOTE:** Be sure to replace `@latest` in the URL with the version of the asset you want. @latest is NOT aware of any MAJOR releases, use at your own risk.
+Include the following example HTML code in the `<head>` element of your page.
 
 ```html
-<link rel="stylesheet" href="https://unpkg.com/@aurodesignsystem/design-tokens@latest/dist/tokens/CSSCustomProperties.css" />
-<link rel="stylesheet" href="https://unpkg.com/@aurodesignsystem/webcorestylesheets@latest/dist/bundled/essentials.css" />
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/design-tokens@latest/dist/tokens/CSSCustomProperties.css" />
```
Comment (blackfalcon) [inline README.md:54]:
```suggestion
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css" />
```

File: README.md [hunk 8]
```diff
@@ -42,24 +42,28 @@ import "@aurodesignsystem/auro-avatar";
...
+<script src="https://cdn.jsdelivr.net/npm/@aurodesignsystem/auro-avatar@latest/dist/auro-avatar__bundled.js" type="module"></script>
+<script src="https://cdn.jsdelivr.net/npm/@aurodesignsystem/auro-avatar@latest/dist/auro-avatar__bundled.es5.js" nomodule></script>
 ```
 
+**NOTE:** Be sure to replace `@latest` in the URL with the version of the asset you want. @latest is NOT aware of any MAJOR releases, use at your own risk.
+
+### polyfills.js
+
+The `polyfills.js` is packaged with this component, but **IT IS NOT NEEDED** to load a polyfill per component. The `polyfills.js` will work for all additional components added to the project.
+
```
Comment (blackfalcon) [inline README.md:66]: Delete this

File: package.json [hunk 9]
```diff
@@ -17,68 +17,62 @@
...
+    "@rollup/plugin-node-resolve": "^15.0.2",
+    "@semantic-release/changelog": "^6.0.3",
     "@semantic-release/git": "^10.0.1",
-    "@semantic-release/npm": "^9.0.2",
-    "@web/dev-server": "^0.1.35",
-    "@web/test-runner": "^0.15.0",
-    "@webcomponents/webcomponentsjs": "^2.7.0",
-    "autoprefixer": "^10.4.13",
+    "@semantic-release/npm": "^10.0.3",
+    "@types/node": "^20.2.4",
```
Comment (blackfalcon) [inline package.json:44]: What is this?

File: scripts/postCss.js [hunk 10]
```diff
@@ -14,44 +11,11 @@ cssFiles.map(name => fs.readFile(`src/${name}.css`, (err, css) => {
...
-      rulesToRemove: {
-        ':root': '*'
-      }
-    }))
     .process(css, { from: `src/${name}.css`, to: `src/${name}.css` })
     .then(result => {
       fs.writeFile(`src/${name}.css`, result.css, () => true)
       if ( result.map ) {
-        fs.writeFile(`src/${name}.map`, String(result.map), () => true)
+        fs.writeFile(`src/${name}.map`, result.map.toString(), () => true)
```
Comment (blackfalcon) [inline scripts/postCss.js:18]: Why is this changed from the standard in the Generator?

File: scripts/pre-commit.js [hunk 11]
```diff
@@ -1,17 +0,0 @@
-'use strict';
```
Comment (blackfalcon) [inline scripts/pre-commit.js:1]: Why was this deleted? See the Generator standard https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/scripts/pre-commit.mjs

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: scripts/postCss.js (hunk 10 above)
Comment (fajar-apri-alaska) [reply scripts/postCss.js:18]: this one also not sure, I assume my IDE on save linters did this, reverting back.

File: package.json (hunk 9 above)
Comment (fajar-apri-alaska) [reply package.json]: not sure, I think it was to solve a problem in typescript installation.
But I tried to remove it and it still can build properly.

**Review by blackfalcon — APPROVED** [review]
